endpoint.go

```diff
@@ -858,3 +861,25 @@ func (c *controller) cleanupLocalEndpoints() {
 		}
 	}
 }
+
+func (ep *endpoint) checkDefaultRoute() error {
+	defaultRoute := 0
+	if len(ep.Gateway()) > 0 {
+		defaultRoute++
+	}
+	for _, route := range ep.joinInfo.StaticRoutes {
+		if route.Destination.String() == "0.0.0.0/0" {
+			defaultRoute++
+		}
+	}
+	if defaultRoute > 1 {
+		return fmt.Errorf("default route conflicts, multiple default route available")
+	}
+	if defaultRoute == 1 {
+		if !ep.joinInfo.disableGatewayService {
+			log.Warn("default route conflicts, disabling libnetwork default gateway service")
```

Comment:
In order to provide proper backward compatibility, I don't think we need this if check and warning.
Especially because this case will be true for bridge driver (without the driver change you implemented in this PR, which I prefer not to add to prove that this change in libnetwork doesn't require any existing driver changes).